client/src/components/Questions/PhotoInput.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useRef } from 'react';
+import { HiOutlinePhoto } from 'react-icons/hi2';
+import { IoCloseCircleSharp } from 'react-icons/io5';
+import { filesize } from 'filesize';
+
+function PhotoInput({
+  files,
+  setFiles,
+}) {
+  const MAX_PHOTOS = 5;
+  const fileInput = useRef([]);
+  const keyTracker = useRef(0);
+
+  const chooseFile = (i) => {
+    keyTracker.current += 1;
+    fileInput.current[i].click();
+  };
+
+  const handleChange = (e, i) => {
+    if (files.length) {
+      const copy = [...files];
+      copy.splice(i, 1, e.target.files[0]);
+      setFiles(copy);
+    } else {
+      setFiles([...files, ...e.target.files]);
+    }
+  };
+
+  const removeFile = (i) => {
+    const copy = [...files];
+    copy.splice(i, 1);
+    setFiles(copy);
+  };
+
+  return (
+    <div className="qa add-photos">
+      {files.map((file, i) => (
+        <div
+          // Exception: Items will never change order, no valid ID.
```

Comment:
Want to clarify if backing out of uploading the file will break this code. I noticed that combining input type="file" and onChange could lead to this bug. The solution seems to be use onLoad or ensure e.target.files[0] exists prior to mapping the thumbnail

Reply:
Demo'd working fix